test/regression/2527_distortion_leak.cc

```diff
@@ -0,0 +1,173 @@
+void OnNewCameraFrame(int* _imageCounter, unsigned char* _imageDest,
+                  const unsigned char *_image,
+                  unsigned int _width, unsigned int _height,
+                  unsigned int _depth,
+                  const std::string &_format)
+{
+  std::lock_guard<std::mutex> lock(mutex);
+  pixelFormat = _format;
+  memcpy(_imageDest, _image, _width * _height * _depth);
+  *_imageCounter += 1;
+}
+
+/////////////////////////////////////////////////
+// \brief Test for issue #1208
+TEST_F(Issue2527Test, Distortion)
+{
+  Load("worlds/issue_2527_camera_distortion.world");
+
+  // Make sure the render engine is available.
+  if (rendering::RenderEngine::Instance()->GetRenderPathType() ==
+      rendering::RenderEngine::NONE)
+  {
+    gzerr << "No rendering engine, unable to run camera test\n";
+    return;
+  }
+
+  sensors::CameraSensorPtr sensorNormal = 
+      std::dynamic_pointer_cast<sensors::CameraSensor>(
+      sensors::get_sensor("camera_normal"));
+  EXPECT_TRUE(sensorNormal != nullptr);
+
+  sensors::CameraSensorPtr sensorBarrel = 
+      std::dynamic_pointer_cast<sensors::CameraSensor>(
+      sensors::get_sensor("camera_barrel"));
+  EXPECT_TRUE(sensorBarrel != nullptr);
+
+  sensors::CameraSensorPtr sensorPincushion = 
+      std::dynamic_pointer_cast<sensors::CameraSensor>(
+      sensors::get_sensor("camera_pincushion"));
+  EXPECT_TRUE(sensorPincushion != nullptr);
+
+  unsigned int width  = 320;
+  unsigned int height = 240;
+
+  imageCountNormal = 0;
+  imageCountBarrel = 0;
+  imageCountPincushion = 0;
+  imgNormal = new unsigned char[width * height*3];
+  imgBarrel = new unsigned char[width * height*3];
+  imgPincushion = new unsigned char[width * height*3];
+
+  event::ConnectionPtr c2 =
+    sensorNormal->Camera()->ConnectNewImageFrame(
+        std::bind(&::OnNewCameraFrame, &imageCountNormal, imgNormal,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+  event::ConnectionPtr c3 =
+    sensorBarrel->Camera()->ConnectNewImageFrame(
+        std::bind(&::OnNewCameraFrame, &imageCountBarrel, imgBarrel,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+  event::ConnectionPtr c4 =
+    sensorPincushion->Camera()->ConnectNewImageFrame(
+        std::bind(&::OnNewCameraFrame, &imageCountPincushion, imgPincushion,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+
+  while (imageCountNormal < 10 || imageCountBarrel < 10 || imageCountPincushion < 10)
+  {
+    common::Time::MSleep(10);
+  }
+
+  unsigned int diffMax = 0, diffSum = 0;
+  double diffAvg = 0.0;
+
+  this->ImageCompare(imgNormal, imgBarrel, width, height, 3,
+                     diffMax, diffSum, diffAvg);
+  EXPECT_NE(diffSum, 0u);
```

Comment:
According with the [official docs](https://google.github.io/googletest/reference/assertions.html): `EXPECT_THAT(value,matcher)`

```suggestion
  EXPECT_NE(0u, diffSum);
```